fuse/data/datasets/dataset_default.py

```diff
@@ -40,6 +40,7 @@ def __init__(
         dynamic_pipeline: Optional[PipelineDefault] = None,
         cacher: Optional[SamplesCacher] = None,
         allow_uncached_sample_morphing: bool = False,
+        always_return_dict = True,
```

Comment:
let's changed the allow_uncached_sample_morphing to mode instead of boolean.
We can have a mode that goes over all the samples to filter Nones or to split samples.
And a mode that skips it and returns whatever the pipeline generates.
See if you can come up with good names for the argument and the modes.